drivers/digital-io/max22190/iio_max22190.c

```diff
@@ -0,0 +1,608 @@
+*******************************************************************************/
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include "no_os_alloc.h"
+#include "no_os_error.h"
+#include "no_os_units.h"
+#include "no_os_util.h"
+
+#include "max22190.h"
+#include "iio_max22190.h"
+
+#define MAX22190_CHANNEL(_addr)			\
+        {					\
+            	.ch_type = IIO_VOLTAGE,		\
+        	.indexed = 1,			\
+		.channel = _addr,		\
+	    	.address = _addr,		\
+	}
+
+static int max22190_iio_read_raw(void *dev, char *buf, uint32_t len,
+				 const struct iio_ch_info *channel,
+				 intptr_t priv);
+
+static int max22190_iio_read_offset(void *dev, char *buf, uint32_t len,
+				    const struct iio_ch_info *channel,
+				    intptr_t priv);
+
+static int max22190_iio_read_scale(void *dev, char *buf, uint32_t len,
+				   const struct iio_ch_info *channel,
+				   intptr_t priv);
+
+static int max22190_iio_read_filter(void *dev, char *buf, uint32_t len,
+				    const struct iio_ch_info *channel,
+				    intptr_t priv);
+
+static int max22190_iio_write_filter(void *dev, char *buf, uint32_t len,
+				     const struct iio_ch_info *channel,
+				     intptr_t priv);
+
+static int max22190_iio_read_filter_delay(void *dev, char *buf, uint32_t len,
+		const struct iio_ch_info *channel,
+		intptr_t priv);
+
+static int max22190_iio_write_filter_delay(void *dev, char *buf, uint32_t len,
+		const struct iio_ch_info *channel,
+		intptr_t priv);
+
+static int max22190_iio_read_filter_available(void *dev, char *buf,
+		uint32_t len,
+		const struct iio_ch_info *channel,
+		intptr_t priv);
+
+static int max22190_iio_read_fault1(void *dev, char *buf, uint32_t len,
+				    const struct iio_ch_info *channel,
+				    intptr_t priv);
+
+static int max22190_iio_read_fault2(void *dev, char *buf, uint32_t len,
+				    const struct iio_ch_info *channel,
+				    intptr_t priv);
+
+static int max22190_iio_reg_read(struct max22190_iio_desc *dev, uint32_t reg,
+				 uint32_t *readval);
+
+static int max22190_iio_reg_write(struct max22190_iio_desc *dev, uint32_t reg,
+				  uint32_t writeval);
+
+static int max22190_iio_setup_channels(struct max22190_iio_desc *desc,
+				       bool *ch_enabled);
+
+static const uint32_t max22190_delay_avail[8] = {
+	50, 100, 400, 800, 1800, 3200, 12800, 20000
+};
+
+static struct iio_attribute max22190_attrs[] = {
+	{
+		.name = "raw",
+		.show = max22190_iio_read_raw,
+	},
+	{
+		.name = "offset",
+		.show = max22190_iio_read_offset,
+	},
+	{
+		.name = "scale",
+		.show = max22190_iio_read_scale,
+	},
+	{
+		.name = "filter_bypass",
+		.show = max22190_iio_read_filter,
+		.store = max22190_iio_write_filter,
+	},
+	{
+		.name = "filter_delay",
+		.show = max22190_iio_read_filter_delay,
+		.store = max22190_iio_write_filter_delay,
+	},
+	{
+		.name = "filter_delay_available",
+		.show = max22190_iio_read_filter_available,
+		.shared = IIO_SHARED_BY_ALL
+	},
+	{
+		.name = "fault1",
```

Comment:
as discussed, this approach is a bit ugly but iio doesn't provide a proper way for exposing the status that comes with the data, so from my point of view it's good, maybe team has a better idea ?

Reply:
Will try this instead.